### PR TITLE
Add a cloneType method to QueueIO

### LINKED ISSUE #245: Vector of Queues wants a cloneType for QueueIO
Chisel3 doesn't like a Vector of Queues, e.g.:
```scala
class QueueBundle extends Bundle {
  val x = Bool
}

class VecOfQueues extends Module {
  val io = new Bundle {
    val x = Bool(INPUT)
    val y = Bool(OUTPUT)
  }

  val voq = Vec(Module(new Queue(new QueueBundle, 2)).io, 2)
}
```
The above example errors out complaining about `QueueIO` needing a `cloneType` method:
```
[error] (run-main-2) chisel3.internal.ChiselException: [error] VecOfQueues.scala:15: Parameterized Bundle class chisel3.util.QueueIO needs cloneType method in class VecOfQueues
[error] [error] VecOfQueues.scala:15: Parameterized Bundle class chisel3.util.QueueIO needs cloneType method in class VecOfQueues
[error] [error] VecOfQueues.scala:15: Parameterized Bundle class chisel3.util.QueueIO needs cloneType method in class VecOfQueues
[error] [error] VecOfQueues.scala:15: Parameterized Bundle class chisel3.util.QueueIO needs cloneType method in class VecOfQueues
[error] CODE HAS 4 ERRORS and 0 WARNINGS
```

Giving `QueueIO` a dedicated `cloneType` method (like `DeqIO` and `EnqIO`) does get it past this, but I can't comment as to whether or not this is the correct approach. I can throw a pull request your way if you want to use it.

### PR DESCRIPTION
**Pull request by seldridge**
Fixes #245.

### CODE REVIEW

File: src/main/scala/chisel3/util/Decoupled.scala [hunk 1]
```diff
@@ -88,6 +88,8 @@ class QueueIO[T <: Data](gen: T, entries: Int) extends Bundle
   val deq   = Decoupled(gen.cloneType)
   /** The current amount of data in the queue */
   val count = UInt(OUTPUT, log2Up(entries + 1))
+
+  override def cloneType: this.type = { new QueueIO(gen, entries).asInstanceOf[this.type]; }
```
Comment (sdtwigg) [inline src/main/scala/chisel3/util/Decoupled.scala]: The ; and {} are unnecessary and do not follow the style of other cloneType methods.

**Comment by seldridge** [comment]
I was naively matching the `DeqIO` and `EnqIO` `cloneType` methods. I went ahead and cleaned those up and the proposed one for `QueueIO`.

**Comment by ucbjrl** [comment]
Ok.

**Comment by ucbjrl** [comment]
ok to test

**Pull request merged**